File: 하은/스택/1874.js
const stack = [];
let top = -1;

function isEmpty() {
    return top === -1;
}

function peek() {
    return top === -1 ? 0 : stack[top];
}
function push(item) {
    stack[++top] = item;
    return;
}

function pop() {
    if (!isEmpty()) {
        return stack[top--];
    }
    return false;
}

const solution = () => {
    const fs = require('fs');
    const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';

    const numbers = fs.readFileSync(filePath, 'utf-8').trim().split('\n').map(Number);

    const res = [];
    let currentNum = 1;

    for (let i = 1; i <= numbers[0]; i++) {
        const num = numbers[i];

        while (currentNum <= num) {
            push(currentNum);
            res.push('+');
            currentNum++;
        }

        if (peek() === num) {
            pop();
            res.push('-');
        } else {
            return 'NO';
        }
    }

    return res.join('\n');
};

console.log(solution().trim());

File: grulla99/stack/1874.py
import sys

input = sys.stdin.readline

n = int(input())

stack = []
result = []
flag = True
now = 1

for i in range(n):
    num = int(input())

    while now <= num:
        stack.append(now)
        result.append('+')
        now += 1

    if stack[-1] == num:
        stack.pop()
        result.append('-')
    else:
        flag = False

if not flag:
    print('NO')
else:
    for i in result:
        print(i)